src/server/sample_data/sample_data_mixin.js

```diff
@@ -0,0 +1,50 @@
+import Joi from 'joi';
+import { dataSetSchema } from './data_set_schema';
+import {
+  createListRoute,
+  createInstallRoute,
+  createUninstallRoute,
+} from './routes';
+import {
+  flightsSpecProvider,
+} from './data_sets';
+
+export function sampleDataMixin(kbnServer, server) {
+  server.route(createListRoute());
+  server.route(createInstallRoute());
+  server.route(createUninstallRoute());
+
+  const sampleDataSets = [];
+
+  server.decorate('server', 'getSampleDataSets', () => {
+    return sampleDataSets;
+  });
+
+  server.decorate('server', 'registerSampleDataSet', (specProvider) => {
+    const { error, value } = Joi.validate(specProvider(server), dataSetSchema);
+
+    if (error) {
+      throw new Error(`Unable to register sample data set spec because its invalid. ${error}`);
+    }
+
+    const defaultIndexSavedObjectJson = value.savedObjects.find(savedObjectJson => {
+      return savedObjectJson.type === 'index-pattern' && savedObjectJson.id === value.defaultIndex;
+    });
+    if (!defaultIndexSavedObjectJson) {
+      throw new Error(
+        `Unable to register sample data set spec, defaultIndex: "${value.defaultIndex}" does not exist in savedObjects list.`);
```

Comment:
I'd do a find/replace for `'data set' -> 'dataset'` and `'dataSet' -> 'dataset'`